client/src/Components/Layouts/Ordersmanagement/index.jsx

```diff
@@ -1,60 +1,284 @@
+    status: "",
+    name: "",
+    error: "",
+    filter: false,
+    stores: []
   };
 
   componentDidMount() {
     axios
       .get("/api/v1//viewOrders")
       .then(res => {
-        this.setState({ orders: res.data });
+        if (res.status === 204) {
+          let error = [...this.state.error];
+          error.response = res;
+          error.response.data = "Error, No orders yet.";
+          this.setState({ error });
+        } else {
+          this.setState({ orders: res.data });
+        }
       })
-      .catch(error => {
+      .catch((error) => {
         this.setState({
           error
         });
       });
+    axios
+      .get("/api/v1/getStores")
+      .then(res => {
+        if (res) {
+          this.setState({ stores: res.data });
+        }
+      })
+      .catch(error => this.setState({ error }));
   }
 
+  dateFilter = object => {
+    if (object) {
+      const { date } = this.state;
+      if (date.length) {
+        if (date[0] && date[1]) {
+          if (date[0].isValid() && date[1].isValid()) {
+            const fromDate = date[0].toDate().setHours(0, 0, 0, 0);
+            const toDate = date[1].toDate().setHours(0, 0, 0, 0);
+            let filtered = object.filter(order => {
+              if (
+                moment(order.date)
+                  .toDate()
+                  .setHours(0, 0, 0, 0) >= fromDate &&
+                moment(order.date)
+                  .toDate()
+                  .setHours(0, 0, 0, 0) <= toDate
+              ) {
+                return true;
+              }
+            });
+            return filtered;
+          }
+        }
+      } else {
+        return object;
+      }
+    }
+  };
+
+  statusFilter = object => {
+    if (object) {
+      const { status } = this.state;
+      if (status) {
+        let filtered = [];
+        const regex1 = new RegExp(/^[(ت)]/);
+        const regex2 = new RegExp(/^[(ل)]/);
+        filtered = object.filter(order => {
+          if (
+            (order.b_status === true &&
+              regex1.test(status) &&
+              "تم".indexOf(status) != -1) ||
+            (order.b_status === false &&
+              regex2.test(status) &&
+              "لم يتم".indexOf(status) != -1)
+          ) {
+            return true;
+          } else if (order.b_status == status) {
+            return true;
+          }
+        });
+        return filtered;
+      } else {
+        return object;
+      }
+    }
+  };
+
+  nameFilter = object => {
+    if (object) {
+      const { name } = this.state;
+      if (name) {
+        let filtered = [];
+        filtered = object.filter(order => {
+          if (order.captain.indexOf(name) != -1) {
+            return true;
+          }
+        });
+        return filtered;
+      } else {
+        return object;
+      }
+    }
+  };
+
+  filter = async (type, value) => {
+    const { date, status, name, orders } = this.state;
+    let filtered = [];
+    if (type === "date") {
```

Comment:
@Anies12 I think this big function might be easier to read and manage if you break it up into three functions (filterDate, filterStatus, filterName) instead of having the big if-statement for `type`

Reply:
you are right, I realized that after I have done this and was already struggling to keep going

Reply:
you are right, I realized that after I did this and I was struggling to keep going :)

Reply:
haha yeah, it sometimes helps to come back to it the next day after getting some sleep, to get a fresh perspective.

Reply:
haha, this is the only thing that I could do